hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/MasterProcedureScheduler.java

```diff
@@ -368,6 +374,13 @@ private static <T extends Comparable<T>> void addToRunQueue(FairQueue<T> fairq,
 
   private static <T extends Comparable<T>> void removeFromRunQueue(FairQueue<T> fairq,
     Queue<T> queue, Supplier<String> reason) {
+    if (hasNoLockNeededProcedure(queue)) {
+      if (LOG.isTraceEnabled()) {
+        LOG.trace("DO NOT remove {} from run queue because There are still procedures in the "
+          + "queue that do not need to acquire locks", queue);
+      }
+      return;
+    }
```

Comment:
Mind explain why we do this here? I see `removeFromRunQueue` is called from several places, some apparently expecting the call to effectively remove the queue.